main.py

```diff
@@ -298,169 +308,210 @@ def menu_live(params):
+    #     return False
+
+    # get media details
+    media = api.get_media_details(mid, is_live)
     common.plugin.log(json.dumps(media))
-    
-    #get media stream URL
+
+    # get media stream URL
     media_url = None
     stream_node = media.get('url_streaming')
-    
+
     if stream_node:
-        #live media streaming
-        if is_live and utils.media_is_streaming(media):
-            media_url = stream_node.get('url_hls','').encode('utf-8').strip()
+        if drm:
+            media_url = api.get_video_detail(mid).get('data').get('urlHlsAes128', '').encode('utf-8').strip()
+
+        # live media streaming
+        elif is_live and utils.media_is_streaming(media):
+            media_url = stream_node.get('url_hls', '').encode('utf-8').strip()
 
-        #regular media
+        # regular media
         else:
-            media_url = stream_node.get('url','').encode('utf-8').strip()
-    
+            media_url = stream_node.get('url', '').encode('utf-8').strip()
+
     if not media_url:
         common.plugin.log_error("unable to get stream URL.")
         common.popup("Impossible de trouver le flux media")
-        return False #TOFIX how to cancel media play ?
-    
+        return False  # TOFIX how to cancel media play ?
+
     common.plugin.log(media_url)
-    #common.popup(media_url)
+    # common.popup(media_url)
 
-    #get auth
+    # get auth
     if drm:
 
-        #get user token
+        # get user token
         try:
             user_token = get_user_jwt_token()
         except ValueError as e:
-            common.popup(e) # warn user
-            return False #TOFIX how to cancel media play ?
-        
-        auth = api.get_drm_media_auth(user_token,mid,is_live)
-        #common.popup("media #{0} auth: {1}".format(mid,auth))
-        
-    #play
-    liz = xbmcgui.ListItem(path=media_url)
+            common.popup(e)  # warn user
+            return False  # TOFIX how to cancel media play ?
+
+        auth = api.get_drm_media_auth(user_token, mid, is_live)
+        # common.popup("media #{0} auth: {1}".format(mid,auth))
+
+        liz = xbmcgui.ListItem(path=media_url)
+
+        # you should activate inputstreamaddon in kodi
+        # and add needed libs (ssd_wv.dll or .so, widevinecdm.dll or .so)
+        # in the cdm folder in special://home.
+        liz.setProperty('inputstreamaddon', 'inputstream.adaptive')
+        liz.setProperty('inputstream.adaptive.license_type', 'com.widevine.alpha')
+        liz.setProperty('inputstream.adaptive.manifest_type', 'hls')
+
+        # Template:  URL|Additional Header|Post-Body|Response Format
+        # Additional Header : customData is auth in base64
+        # Post-Body: R{SSM} -> Raw SendSessionMessage (== Raw Bytes in Post Body)
+        # Response format: by default: expect raw bytes in response payload -> leave empty
+
+        licence_url = 'http://wv-keyos.licensekeyserver.com|customData=' + auth + '|R{SSM}|'
```

Comment:
Maybe we should dynamically get the license key server :

        # Get DRM url from settings
        app_datas = api.get_app_settings()
        licence_url = app_datas['media_drm_url']['widevine_url']

        licence_url = licence_url + '|customData=' + auth + '|R{SSM}|'